### PR TITLE
Python3 - re-apply on top of Ormod

### PR DESCRIPTION
**Pull request by redref**
Tried to get things going on top of previous PR #259 . 

Fixed dependencies 
  * mysql : MySQLdb on python2, mysqlclient on python 3
  * users : pyutmp on python2, utmp on python3 (utmp may fit both)
  * lmsensors : python3 required library come from a bitbucket repo, I did'nt included in travis (test skipped), but tests are working on my host.

TOFIX
  * beanstalkc : I did'nt find suitable library on py3k
  * bernhard (protobuf) : pypi require non-py3k ready protobuf, commited a workaround. Any better solution ?

Fixed tests
  * kafkastat : test was failing before, had to rework on collector to make sure py3k is not source. I cannot explain how this test was working before.

My previous PR #409 is included. I saw it quite late. If necessary, I can rework tree to get it removed.

### CODE REVIEW

**Comment by mattrobenolt** [comment]
At a quick glance, none of my concern that I pointed out on #259 were addressed in these additions. There was lots of change in behavior for python 2 that wouldn't be ideal.

**Comment by mattrobenolt** [comment]
Oh, I take that back. These commit messages are extremely misleading. I'll look through this again.

File: Makefile [hunk 1]
```diff
@@ -61,23 +62,29 @@ buildrpm: sdist
...
 deb: builddeb
 
 sdeb: buildsourcedeb
 
-builddeb: version 
+builddeb: version
 	dch --newversion $(VERSION) --distribution unstable --force-distribution -b "Last Commit: $(shell git log -1 --pretty=format:'(%ai) %H %cn <%ce>')"
 	dch --release  "new upstream"
-	./setup.py sdist --prune
+	./setup.py sdist
```
Comment (mattrobenolt) [inline Makefile]: Why did all the other setup.py calls change to `$(PYTHON)` but these don't?

File: bin/diamond-setup [hunk 2]
```diff
@@ -77,7 +78,7 @@ def getCollectors(path):
 
 
 def typeToString(key):
-    if isinstance(obj.config[key], basestring):
+    if isinstance(obj.config[key], str):
```
Comment (mattrobenolt) [inline bin/diamond-setup]: This changed meaning for py2. `basestring == (str, unicode)`

File: src/collectors/conntrack/conntrack.py [hunk 3]
```diff
@@ -50,12 +50,12 @@ def collect(self):
         collected = {}
         files = []
 
-        if isinstance(self.config['dir'], basestring):
+        if isinstance(self.config['dir'], str):
```
Comment (mattrobenolt) [inline src/collectors/conntrack/conntrack.py]: Changed behavior.

File: src/collectors/darner/darner.py [hunk 4]
```diff
@@ -117,7 +118,7 @@ def collect(self):
         hosts = self.config.get('hosts')
 
         # Convert a string config value to be an array
-        if isinstance(hosts, basestring):
+        if isinstance(hosts, str):
```
Comment (mattrobenolt) [inline src/collectors/darner/darner.py]: Changed behavior.

File: src/collectors/darner/darner.py [hunk 5]
```diff
@@ -140,7 +141,7 @@ def collect(self):
                             queues[queue][queue_stat])
 
             # figure out what we're configured to get, defaulting to everything
-            desired = self.config.get('publish', stats.keys())
+            desired = self.config.get('publish', iterkeys(stats))
```
Comment (mattrobenolt) [inline src/collectors/darner/darner.py]: This might be ok, but this changed from a list to an iterator.

**Comment by mattrobenolt** [comment]
I have to get back to this later, but there's still quite a bit of changed behaviors left. basestrings changed to str and iterators changed to lists, lists changed to iterators, etc.

**Comment by redref** [comment]
@mattrobenolt I made a review on my side and corrected basetring -> str on some remarks.

Sorry about commit messages. I lack some github experiences.

**Comment by redref** [comment]
Hi. Looking commit history and what has to be done, I think this can be done in more incremental way.

Something like 3/4 steps
  * changes on libraries (and Kafka check)
  * changes on syntax
  * changes on build/checks systems

Let me know your preference on this.

**Comment by eirnym** [comment]
Why this PR was closed?

**Pull request closed without merging**